doc/tutorials/content/gpu_octree.rst

```diff
@@ -0,0 +1,235 @@
+
+Note that unlike in the CPU octree implementation, the user cannot set the resolution of the octree. Instead, the depth of the octree will be automatically determined depending on the point distribution of the cloud (However, the maximum depth is capped at 10).
+
+.. literalinclude:: sources/gpu_octree/gpu_octree.cpp
+    :language: cpp
+    :lines: 47-58
+
+After building the octree, we we can move on to the different search queries.
+
+K Nearest Neighbors search
+---------------------------
+
+The K nearest neighbor algorithm is designed to identify a given (k) number of nearest neighbors to a query point. However, the GPU KNN search algorithm is currently restricted to k=1 only.
+
+
+.. note::
+
+    The GPU K nearest neighbor algorithm differs from it's CPU counterpart in implementation. Specifically, the CPU algorithm will only find approximate nearest neighbors, where the user can set the upper bound of the error allowed. In contrast, the GPU algorithm finds the absolute nearest neighbor.
+
+In order to perform K nearest search, we must prepare output buffers on the device to store the results of the search queries. Specifically, we need two separate buffers to store the resulting nearest neighbor indices and the corresponding squared distances between the search point and nearest neighbors. Note that these buffers will be automatically resized to `k * number of queries` size, in order to hold all the results.
+
+Now, we can perform the K nearest neighbors search on GPU. After performing the search we must download the results from GPU memory into host memory using two new vectors for indices and square distances.
+
+
+.. literalinclude:: sources/gpu_octree/gpu_octree.cpp
+    :language: cpp
+    :lines: 62-98
+
+Radius Search
+-------------
+
+Radius search returns neighbors within a given radius to the query point. The user must specify a maximum limit of points to return per query. There are implementations radius search available which either return / do not return the corresponding squared distances to the query point. Note that the methods which do not return squared distances are more efficient than those that do. In usage, this works very similar to the K nearest neighbor search.
+
+Neighbors are returned as a `pcl::gpu::NeighborIndices` object, which contains all result indices packed into a one dimensional vector called `data`, as well as an additional vector called `sizes` which contains the number of results found for each individual query. These sizes can can be used to split the resulting indices as well as square distances by query.
+
+Radius search can be conducted with a single radius for all queries or, individual radii may be specified for each query, by passing a `pcl::gpu::Octree::Radiuses` object of the same size as the queries object to the radius search method. In this tutorial we will specify individual radii for each of our ten queries.
```

Comment:
```suggestion
Radius search can be conducted with a single radius for all queries or, individual radius may be specified for each query, by passing a `pcl::gpu::Octree::Radiuses` object of the same size as the queries object to the radius search method. In this tutorial we will specify individual radius for each of our ten queries.
```

Reply:
I really think he meant the plural of radius here.

Reply:
I thought it would be like:

X number of queries shares the same radius
or
Each query have exactly one radius each

That's why I would think it should be a single radius (ie. individual radius, for each query) - even though you are supplying multiple radii (the array), but one for each individual query.

You are not supplying multiple radii to a single query and that's what I'm reading in the current form?

Reply:
Yeah, I can see what you mean. `radius` may be clearer for the reader.